src/main/java/org/logstash/dissect/DelimiterFactory.java

```diff
@@ -0,0 +1,20 @@
+package org.logstash.dissect;
+
+import org.logstash.dissect.search.DoubleByteSearchStrategy;
+import org.logstash.dissect.search.MultiByteSearchStrategy;
+import org.logstash.dissect.search.SingleByteSearchStrategy;
+import org.logstash.dissect.search.ZeroByteSearchStrategy;
+
+class DelimiterFactory {
+    public static Delimiter create(String delim) {
+        if (delim.length() == 1) {
+            return new Delimiter(delim).addStrategy(new SingleByteSearchStrategy());
```

Comment:
All implementations of `DelimiterSearchStrategy` are stateless so there is no need to create additional garbage. The usual idiom in Java is something along these lines:

```java
final class JustOneInstanceNeeded {
    public static final JustOneInstanceNeeded INSTANCE = new JustOneInstanceNeeded();

    private JustOneInstanceNeeded() {
    }

    public int foo() {
    }
}